nbbuild/cluster.properties

```diff
@@ -299,8 +298,7 @@ nb.cluster.platform=\
         spi.actions,\
         spi.quicksearch,\
         templates,\
-        templatesui,\
-        uihandler
```

Comment:
Both [lib.uihandler](https://bits.netbeans.org/12.6/javadoc/org-netbeans-lib-uihandler/org/netbeans/lib/uihandler/package-summary.html) as well as [uihandler](https://bits.netbeans.org/12.6/javadoc/org-netbeans-modules-uihandler/overview-summary.html) are an API of 12.6.

We should keep them around as autoload modules, in case some plugin or NetBeans application continues to use them. Disabled autoloads have close to no overhead on the runtime performance of the system and given NetBeans project commitment to compatibility (or is it mine?) I'd rather kept them around.

Reply:
i had that thought too.

If I revert the changes in the property files, set the modules to `autoload=false` in manifest but keep the dependency change in ide.branding.kit/nbproject/project.xml. I think we can be backwards compatible while having stats and exception reporting not loaded by default. Would that be ok?

Reply:
update: it still loads without autoload

is there a way to debug which module is responsible for loading a particular other module?

Reply:
I usually put conditional breakpoint into `org.netbeans.Module.setEnabled(...)` to see who's enabling the module.

Reply:
i can't explain why its getting loaded on startup.
```java
    void setEnabled(boolean enabled) {
        /* #13647: actually can happen if loading of bootstrap modules is rolled back: */
        if (isFixed() && ! enabled) throw new IllegalStateException("Cannot disable a fixed module: " + this); // NOI18N
        if (!this.isEnabled()) {
            Dependency[] deps = this.data.getDependencies();
            if(deps != null) {
                for (Dependency dependency : deps) {
                    if (dependency.getName().contains("uihandler") || dependency.getName().contains("exceptionreporter")) {
                        System.out.println(this.data.getCodeName()+" (autoload: "+isAutoload()+", eager: "+isEager()+")");
                        System.out.println("depends on: "+dependency);
                        System.out.println();
                    }
                }
            }
        }
        this.enabled = enabled;
    }
```
results in:
```
org.netbeans.modules.uihandler (autoload: false, eager: false)
depends on: module org.netbeans.lib.uihandler > 1.37

org.netbeans.modules.uihandler.exceptionreporter (autoload: false, eager: false)
depends on: module org.netbeans.modules.uihandler > 2.0

org.netbeans.modules.bugzilla.exceptionreporter (autoload: false, eager: true)
depends on: module org.netbeans.lib.uihandler > 1.37
```
nothing depends on the uihandler modules which is also autoloading.

Reply:
`eager: true` - means to enable when possible - e.g. when all dependencies are satisfied. Is the `org.netbeans.modules.bugzilla.exceptionreporter` enabled in runtime? Than it enables also its autoload dependencies. Probably make the module `!eager && autoload`.

Reply:
That module (bugzilla.exceptionreporter) also provides the action in the Help menu to open the issues reporting page on the website I think. Can we remove the dependency instead?

Reply:
question: I see xmls like:
```xml
<module name="org.netbeans.modules.uihandler">
    <param name="autoload">false</param>
    <param name="eager">false</param>
    <param name="enabled">true</param>      <------- 
    <param name="jar">modules/org-netbeans-modules-uihandler.jar</param>
    <param name="reloadable">false</param>
</module>
```
(netbeans/platform/config/Modules)
what in the build is influencing if its "enabled" or not? Some don't have that param. I suppose that is what you mean by "enabled by the runtime"?

@neilcsmith-net that action could be potentially moved into a different module

Reply:
answer:
`is.enabled=false`

that was easy :)

Reply:
@mbien absolutely, although not sure that module does much else now. Either way, please don't lose the action within the context of these changes.

Reply:
modules are now disabled and set to autoload + report issue action moved to "user utilities" (ide/utilities). This is the same module which is also providing the "docs and support" link and the kb shortcuts in the help menu.

Reply:
@neilcsmith-net we could probably remove the `org.netbeans.modules.bugzilla.exceptionreporter` module and `NBBugzillaUtils` in the bugzilla module. Unless you are planning to migrate back to bugzilla in about a year :)

But even the Kenai integration is still there... probably better to leave that to other PRs.